docs/src/app/(public)/(content)/react/utils/use-render/page.mdx

```diff
@@ -0,0 +1,110 @@
+# useRender
+
+<Subtitle>Hook for enabling a render prop in custom components.</Subtitle>
+
+<Meta name="description" content="Hook for enabling a render prop in custom components." />
+
+The `useRender` hook enables custom components to provide a `render` prop to override the default rendered element.
+
+## API reference
+
+### Input parameters
+
+<PropsReferenceTable
+  data={{
+    render: {
+      type: 'RenderProp<State>',
+      description:
+        'The React element or a function that returns one to override the default element.',
+    },
+    props: {
+      type: 'Record<string, unknown>',
+      description:
+        'Props to be spread on the rendered element.\n\nThey are merged with the internal props\n of the component, so that event handlers are merged,\nclass names and styles are joined, and other external props overwrite the internal ones.',
+    },
+    state: {
```

Comment:
Should be `refs`*? Wonder if it's confusing that `params.props.ref` and params.refs` are both a thing?

Reply:
I wanted to make it simple for both scenarios, e.g. ref is already in the props and people don't have additional regs, they don't need to specify anything, if they have additional refs, they can use the refs array.

Reply:
```suggestion
    refs: {
```
(`state` is duplicated below)